src/components/RichText.tsx

```diff
@@ -2,62 +2,87 @@
 import React from 'react';
 import { render, StoryblokRichtext } from 'storyblok-rich-text-react-renderer-ts';
 import { dcnb } from 'cnbuilder';
-import { Heading, Paragraph } from './Typography';
+import { CtaLink } from './Cta';
+import { SbCta } from './Storyblok/SbCta';
+import {
+  Heading,
+  FontSizeType,
+  Paragraph,
+  textAligns,
+  TextAlignType,
+} from './Typography';
 
 export type RichTextProps = {
   wysiwyg: StoryblokRichtext;
-  isDark?: boolean;
+  isLightText?: boolean;
+  textAlign?: TextAlignType;
   className?: string;
   linkColor?: string;
 };
 
 export const RichText = ({
   wysiwyg,
-  isDark,
+  isLightText,
+  textAlign,
   className,
 }: RichTextProps) => {
-  let textColor = 'su-text-current';
+  const textColor = isLightText ? 'su-text-white print:su-text-black' : 'su-text-black';
 
-  if (isDark) {
-    textColor = 'su-text-black-20 print:su-text-black';
-  }
   const rendered = render(wysiwyg, {
     markResolvers: {
-      styled: (children, props) => React.createElement(
-        'span',
-        {
-          // eslint-disable-next-line react/destructuring-assignment
-          className: props?.class || '',
-        },
-        children,
+      styled: (children, props) => (
+        <span className={props.class}>{children}</span>
       ),
       bold: (children) => <strong>{children}</strong>,
       italic: (children) => <em>{children}</em>,
+      link: (children, props) => {
+        const {
+          href,
+          target,
+          linktype,
+          anchor,
+        } = props;
+        // Structure the link data so it takes the same shape as sbLink
+        const sbLink = {
+          linktype,
+          cached_url: linktype !== 'email' ? href : '',
+          email: linktype === 'email' ? href : '',
+          anchor,
+          // The WYSIWYG link adds a target="_self" by default which is unnecessary
+          target: target === '_blank' ? '_blank' : undefined,
+        };
+
+        return (
+          <CtaLink
+            sbLink={sbLink}
+            variant={isLightText ? 'inlineDark' : 'inline'}
+            className="children:su-inline"
+            // Custom link attributes are not supported by the rich text renderer currently
+            // Adding rel="noopener" for all eternal links for security reasons
+            rel={linktype === 'url' ? 'noopener' : undefined}
+          >
+            {children}
+          </CtaLink>
+        );
```

Comment:
The link object inside WYSIWYG is slightly different from the link field in a regular SB component -__-. Here I'm using the props from the WYSIWYG link and structure an object that takes on the shape of a regular SB link so we can pass that into our CtaLink component.

As mentioned in the comment, the current version of storyblok rich text renderer doesn't support the newish "custom" prop from the WYSIWYG link yet, so for now I'm manually adding a rel="noopener" for external links that are added in the WYSIWYG.